library/azure_rm_cosmosdbaccount.py

```diff
@@ -0,0 +1,479 @@
+                type='str',
+                choices=['global_document_db',
+                         'mongo_db',
+                         'parse']
+            ),
+            consistency_policy=dict(
+                type='dict'
+            ),
+            geo_rep_locations=dict(
+                type='list'
+            ),
+            database_account_offer_type=dict(
+                type='str'
+            ),
+            ip_range_filter=dict(
+                type='str'
+            ),
+            is_virtual_network_filter_enabled=dict(
+                type='str'
+            ),
+            enable_automatic_failover=dict(
+                type='bool'
+            ),
+            capabilities=dict(
+                type='list'
+            ),
+            virtual_network_rules=dict(
+                type='list'
+            ),
+            enable_multiple_write_locations=dict(
+                type='str'
+            ),
+            state=dict(
+                type='str',
+                default='present',
+                choices=['present', 'absent']
+            )
+        )
+
+        self.resource_group = None
+        self.name = None
+        self.parameters = dict()
+
+        self.results = dict(changed=False)
+        self.mgmt_client = None
+        self.state = None
+        self.to_do = Actions.NoAction
+
+        super(AzureRMDatabaseAccounts, self).__init__(derived_arg_spec=self.module_arg_spec,
+                                                      supports_check_mode=True,
+                                                      supports_tags=True)
+
+    def exec_module(self, **kwargs):
+        """Main module execution method"""
+
+        for key in list(self.module_arg_spec.keys()) + ['tags']:
+            if hasattr(self, key):
+                setattr(self, key, kwargs[key])
+            elif kwargs[key] is not None:
+                if key == "location":
+                    self.parameters["location"] = kwargs[key]
+                elif key == "kind":
+                    ev = kwargs[key]
+                    if ev == 'global_document_db':
+                        ev = 'GlobalDocumentDB'
+                    elif ev == 'mongo_db':
+                        ev = 'MongoDB'
+                    elif ev == 'parse':
+                        ev = 'Parse'
+                    self.parameters["kind"] = ev
+                elif key == "consistency_policy":
+                    self.parameters["consistency_policy"] = _snake_to_camel(kwargs[key])
+                elif key == "geo_rep_locations":
+                    locations = kwargs[key]
+                    for i in range(len(locations)):
+                        locations[i]['location_name'] = locations[i].pop('name')
+                    self.parameters["locations"] = locations
+                elif key == "database_account_offer_type":
+                    self.parameters["database_account_offer_type"] = kwargs[key]
+                elif key == "ip_range_filter":
```

Comment:
what about a else to cover all others? instead of hardcode everything?

Reply:
that's also solved in different way